modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -369,15 +416,120 @@ protected void doInitialise() throws InitialisationException {
+              .flatMapMany(ctx -> from(publisher)
+                  .doOnNext(event -> prepareAndExecuteOperation(event,
+                                                                () -> {
+                                                                  OperationExecutionParams oep =
+                                                                      ((SdkInternalContext) ((InternalEvent) event)
+                                                                          .<SdkInternalContext>getSdkInternalContext())
+                                                                              .getOperationExecutionParams();
+                                                                  return oep.getCallback();
+                                                                },
+                                                                ctx)));
+        }
+
+        @Override
+        public boolean isAsync() {
+          return ComponentMessageProcessor.this.isAsync();
+        }
+      };
+
+      final Flux<CoreEvent> transformed = from(p)
+          .transform(processingStrategy.onProcessor(operationInnerProcessor))
+          .onErrorContinue((t, event) -> LOGGER.error("Unhandler error in operation (" + toString() + ") flux",
+                                                      t));
+      return from(processingStrategy.registerInternalFlux(transformed));
+    },
+                                                getRuntime().availableProcessors());
+  }
+
+  private CoreEvent addContextToEvent(CoreEvent event, Context ctx) throws MuleException {
+    SdkInternalContext sdkInternalContext =
+        (SdkInternalContext) ((InternalEvent) event).<SdkInternalContext>getSdkInternalContext();
+    if (((InternalEvent) event).getSdkInternalContext() == null) {
+      sdkInternalContext = new SdkInternalContext();
+      ((InternalEvent) event).setSdkInternalContext(sdkInternalContext);
+    }
+
+    if (hasChainNested
+        && (ctx.hasKey(POLICY_NEXT_OPERATION) || ctx.hasKey(POLICY_IS_PROPAGATE_MESSAGE_TRANSFORMATIONS))) {
+      sdkInternalContext.setInnerChainSubscriberContextMapping(innerChainCtx -> {
+        if (ctx.hasKey(POLICY_NEXT_OPERATION)) {
+          innerChainCtx = innerChainCtx.put(POLICY_NEXT_OPERATION, ctx.get(POLICY_NEXT_OPERATION));
+        }
+        if (ctx.hasKey(POLICY_IS_PROPAGATE_MESSAGE_TRANSFORMATIONS)) {
+          innerChainCtx = innerChainCtx.put(POLICY_IS_PROPAGATE_MESSAGE_TRANSFORMATIONS,
+                                            ctx.get(POLICY_IS_PROPAGATE_MESSAGE_TRANSFORMATIONS));
+        }
+        return innerChainCtx;
+      });
+    }
+
+    sdkInternalContext.setConfiguration(resolveConfiguration(event));
+    final Map<String, Object> resolutionResult = getResolutionResult(event, sdkInternalContext.getConfiguration());
+    sdkInternalContext.setResolutionResult(resolutionResult);
+    sdkInternalContext.setPolicyToApply(getLocation() != null
```

Comment:
where is this context cleared/replaced?

Reply:
this is create above in this method. Will add a clear for it